internal/provider/models/alerts/base_model.go

```diff
@@ -0,0 +1,207 @@
+	},
+	"operation": schema.StringAttribute{
+		Required: true,
+		Description: "Specifies the type of aggregation operation to use with the window type and duration. " +
+			"This value must be `custom` for a Log event type.",
+		Validators: []validator.String{
+			stringvalidator.OneOf(MapKeys(Aggregate_Operations)...),
+		},
+	},
+	"window_type": schema.StringAttribute{
+		Optional: true,
+		Computed: true,
+		Description: "Sliding windows can overlap, whereas tumbling windows are disjoint. " +
+			"For example, a tumbling window has a fixed time span and any events that " +
+			"fall within the \"window duration\" will be used in the aggregate. " +
+			"In a sliding window, the aggregation occurs every \"window duration\" seconds " +
+			"after an event is encountered.",
+	},
+	"window_duration_minutes": schema.Int64Attribute{
+		Optional:    true,
+		Computed:    true,
+		Description: "The duration of the aggregation window in minutes.",
+	},
+	"script": schema.StringAttribute{
+		Optional: true,
+		Description: "A custom JavaScript function that will control the aggregation. At the " +
+			"time of flushing, this aggregation will become the emitted event. This script " +
+			"is required when choosing a `custom` operation.",
+	},
+	"event_timestamp": schema.StringAttribute{
+		Optional: true,
+		Description: "The path to a field on the event that contains an epoch timestamp " +
+			"value. If an event does not have a timestamp field, events will be associated " +
+			"to the wall clock value when the event is processed. " +
+			"Required for Log event types and disallowed for Metric event types. ",
+	},
+	"severity": schema.StringAttribute{
+		Optional:    true,
+		Computed:    true,
+		Description: "The severity level of the alert.",
+		Validators: []validator.String{
+			stringvalidator.OneOf("INFO", "WARNING", "MAJOR", "CRITICAL"),
+		},
+		Default: stringdefault.StaticString("INFO"),
+	},
+	"style": schema.StringAttribute{
+		Optional:    true,
+		Computed:    true,
+		Description: "Configuration for how the alert message will be constructed.",
+		Validators: []validator.String{
+			stringvalidator.OneOf("static"),
+		},
+		Default: stringdefault.StaticString("static"),
+	},
+	"subject": schema.StringAttribute{
+		Required:    true,
+		Description: "The subject line to use when the alert is sent.", // TODO: Add details about templates when added
+	},
+	"message": schema.StringAttribute{
+		Required:    true,
+		Description: "The message body to use when the alert is sent.", // TODO: Add details about templates when added
+	},
+	"ingestion_key": schema.StringAttribute{
+		Required:    true,
+		Description: "The key required to ingest the alert into Log Analysis",
+	},
+}
+
+type CheckedFields struct {
+	Operation      StringValue
+	EventType      StringValue
+	Script         StringValue
+	EventTimestamp StringValue
+	GroupBy        ListValue
+}
```

Comment:
Even though these fields would all exist for every alert type model, Go was being a pain about its type system.  Generics can't be used because they don't allow field access.  Again, this appeared to be the "Go way" to do this even though it feels Mickey Mouse to me.